bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/SortedLedgerStorage.java

```diff
@@ -144,9 +144,7 @@ public boolean ledgerExists(long ledgerId) throws IOException {
         // the O(1) for the ledgerCache.
         if (!interleavedLedgerStorage.ledgerExists(ledgerId)) {
             EntryKeyValue kv = memTable.getLastEntry(ledgerId);
-            if (null == kv) {
-                return interleavedLedgerStorage.ledgerExists(ledgerId);
```

Comment:
As the comment said, checking the skip list is an O(logN) operation compared to the O(1) for the ledgerCache.

Which means the `memTable.getLastEntry(ledgerId)` is time consuming operation. After check that, the ledgerCache may be filled so we check it again.